.github/workflows/deploy-catalog.yaml

```diff
@@ -0,0 +1,56 @@
+name: Deploy catalog to ECR
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/deploy-catalog.yaml'
+      - 'catalog/**'
+      - 'shared/**'
+
+jobs:
+  deploy-catalog-ecr:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: catalog
+    env:
+      NODE_OPTIONS: --max-old-space-size=4096
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '16.11'
+          cache: 'npm'
+          cache-dependency-path: 'catalog/package-lock.json'
```

Comment:
i'm not sure this is useful. afaict it's supposed to save the npm cache, which holds the packages fetched from the registry, not `node_modules`, and that means we shouldn't invalidate it on lockfile changes, bc even if several out of thousands of packages have changed, we would benefit hugely from still getting the majority of the packages from cache. it may get somewhat bloated over time, but i guess we can reset it somehow from time to time, or just ignore that, since i believe the overhead would be negligible.
(tho i'm not quite sure it would work properly without this option specified)

Reply:
I don't think I actually understand this. Without this option, it won't cache anything at all, right?

Also, do we want to keep this in `js-ci.yml`? What's the difference?

Reply:
> I don't think I actually understand this. Without this option, it won't cache anything at all, right?

i'm saying we should keep `cache: 'npm'`, but drop `cache-dependency-path`

Reply:
> even if several out of thousands of packages have changed, we would benefit hugely from still getting the majority of the packages from cache. it may get somewhat bloated over time, but i guess we can reset it somehow from time to time, or just ignore that, since i believe the overhead would be negligible.

I think we should use what's recommended unless we understand all consequences e.g. I'm not sure if we pay for the cache, if there is a way to reset cache etc.

https://github.com/actions/setup-node?tab=readme-ov-file#caching-global-packages-data

> The action defaults to search for the dependency file (package-lock.json, npm-shrinkwrap.json or yarn.lock) in the repository root, and uses its hash as a part of the cache key. Use cache-dependency-path for cases when multiple dependency files are used, or they are located in different subdirectories.

Reply:
yeah it doesn't work w/o that key, so im adding it back, sorry: #3880